crates/ruff_python_formatter/src/other/f_string_element.rs

```diff
@@ -204,18 +204,13 @@ impl Format<PyFormatContext<'_>> for FormatFStringExpressionElement<'_> {
             // before the closing curly brace is not strictly necessary, but it's
             // added to maintain consistency.
             let bracket_spacing =
-                match left_most(expression, comments.ranges(), f.context().source()) {
-                    Expr::Dict(_) | Expr::DictComp(_) | Expr::Set(_) | Expr::SetComp(_) => {
-                        Some(format_with(|f| {
-                            if self.context.can_contain_line_breaks() {
-                                soft_line_break_or_space().fmt(f)
-                            } else {
-                                space().fmt(f)
-                            }
-                        }))
+                needs_bracket_spacing(expression, f.context()).then_some(Some(format_with(|f| {
+                    if self.context.can_contain_line_breaks() {
+                        soft_line_break_or_space().fmt(f)
+                    } else {
+                        space().fmt(f)
                     }
-                    _ => None,
-                };
+                })));
```

Comment:
Should we use `.then` here to avoid eager evaluation? Regardless, I don't think we need to wrap it in `Some(..)`.

```rs
            let bracket_spacing = needs_bracket_spacing(expression, f.context()).then(|| {
                format_with(|f| {
                    if self.context.can_contain_line_breaks() {
                        soft_line_break_or_space().fmt(f)
                    } else {
                        space().fmt(f)
                    }
                })
            });
```

Reply:
Oh yeah, the `Some` isn't necessary. I think `then_some` is fine because `format_with` is very cheap, it roughly boils down to returning a function pointer that then gets assigned to `bracket_spacing`. Calling a function is probably more expensive